components/wallet/lnbits.js

```diff
@@ -0,0 +1,124 @@
+import { bolt11Tags } from '@/lib/bolt11'
+
+export const name = 'lnbits'
+
+export const fields = [
+  {
+    name: 'url',
+    label: 'lnbits url',
+    type: 'text'
+  },
+  {
+    name: 'adminKey',
+    label: 'admin key',
+    type: 'password'
+  }
+]
+
+export const card = {
+  title: 'LNbits',
+  badges: ['send only', 'non-custodialish']
+}
+
+export async function validate ({ logger, ...config }) {
+  return await getInfo({ logger, ...config })
+}
+
+async function getInfo ({ logger, ...config }) {
+  const response = await getWallet(config.url, config.adminKey)
+  return {
+    node: {
+      alias: response.name,
+      pubkey: ''
+    },
+    methods: [
+      'getInfo',
+      'getBalance',
+      'sendPayment'
+    ],
+    version: '1.0',
+    supports: ['lightning']
+  }
+}
+
+export async function sendPayment ({ bolt11, config, logger }) {
+  const { url, adminKey } = config
+
+  const hash = bolt11Tags(bolt11).payment_hash
+  logger.info('sending payment:', `payment_hash=${hash}`)
+
+  try {
+    const response = await postPayment(url, adminKey, bolt11)
+
+    const checkResponse = await getPayment(url, adminKey, response.payment_hash)
+    if (!checkResponse.preimage) {
+      throw new Error('No preimage')
+    }
+
+    const preimage = checkResponse.preimage
+    logger.ok('payment successful:', `payment_hash=${hash}`, `preimage=${preimage}`)
+    return { preimage }
+  } catch (err) {
+    logger.error('payment failed:', `payment_hash=${hash}`, err.message || err.toString?.())
+    throw err
+  }
+}
```

Comment:
I know this is super WIP, but just in case, one of the benefits of unifying the interface should be that all the logging and generic error handling can go in `wallet/index.js`

Reply:
Ah yes, now only `validate` logs messages since I think they should be custom per wallet and not generic. Every other logging is handled in `wallet/index.js`.